client/components/signup-form/social.jsx

```diff
@@ -0,0 +1,48 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import GoogleLoginButton from 'components/social-buttons/google';
+import FacebookLoginButton from 'components/social-buttons/facebook';
+
+/**
+ * External dependencies
+ */
+import config from 'config';
+
+class SocialSignupForm extends Component {
+	constructor() {
+		super();
+		this.onGoogleResponse = this.onGoogleResponse.bind( this );
+		this.onFacebookResponse = this.onFacebookResponse.bind( this );
+	}
+
+	onGoogleResponse( response ) {
+		if ( ! response.Zi || ! response.Zi.id_token ) {
+			return;
+		}
+		// TODO: post response to the new wpcom endpoint to login
+	}
+
+	onFacebookResponse( response ) {
+		if ( ! response.email ) {
+			return;
+		}
+		// TODO: post response to the new wpcom endpoint to login
+	}
```

Comment:
It's not "authenticating with google" though, it's the result of the authentication, after the flow is over. We will need to call the wpcom endpoint to authenticate using google `id_token` and we might indeed name the function called here `authenticateWithGoogle`.